File: Moderate/FirstNonRepeat.java
// Eden Ghirmai, 2/18/14, www.codeeval.com
// Find the first non repeated character in a string

import java.io.*;
import java.util.*; 

public class FirstNonRepeat {
    public static void main(String[] args) throws FileNotFoundException {
		File file = new File(args[0]);
		Scanner in = new Scanner(file);
		
		while(in.hasNextLine()) {
			String result = ""; 
			String line = in.nextLine(); 
			for (int i = 0; i < line.length(); i++) {
				String c = line.charAt(i) + ""; 
				if (!line.substring(0, i).contains(c) && !line.substring(i + 1).contains(c)) {
					result = c; 
					break;
				}
			}
			System.out.println(result); 
		}
    }
}

File: Moderate/FirstNonRepeat.py
# Eden Ghirmai, 3/09/14, www.codeeval.com
# Finds the first non repeated character in a string.

import sys

def check(phrase):
    for x in range(0, len(phrase)):
		c = test[x]
		before = test[:x]
		after = test[x+1:]
		if (before.find(c) == -1 and after.find(c) == -1):
			return c
    return         


test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    if test: 
    	print check(test)
    		
test_cases.close()
